ONSdigital/design-system
# Remove undefined checks

### What is the context of this PR?
Resolves issue #2357 

### How to review
1. Verify that tests pass with `yarn test`.
2. Verify that documentation pages render properly.
3. Verify that components still work with Jinja2.
4. 
### Notes
The majority of the cases were automatically replaced with the following script which finds all occurrences that match the pattern `foo.bar is defined and foo.bar` where foo.bar matches on both sides:
```js
const fs = require("fs-extra");
const glob = require("glob").sync;

async function processFile(filePath) {
  let fileContents = await fs.readFile(filePath, { encoding: "utf8" });
  fileContents = fileContents.replace(/(([a-zA-Z_]+[a-zA-Z0-9_]*)(\.([a-zA-Z_]+[a-zA-Z0-9_]*))*) is defined and (([a-zA-Z_]+[a-zA-Z0-9_]*)(\.([a-zA-Z_]+[a-zA-Z0-9_]*))*)/g, (a, b, c, d, e, f, g, h) => {
    return b === f ? b : a;
  });
  await fs.writeFile(filePath, fileContents, { encoding: "utf8" });
}

(async () => {
  await Promise.all(glob("**/*.njk").map(processFile));
})();
```
And then all remaining cases where `is defined` was used were updated by hand.

There were also several odd occurrences of the form `foo.bar and foo.bar is defined`.
